source/atomics.tex

```diff
@@ -791,14 +791,15 @@
 
 \pnum
 \begin{note}
-Hardware could require an object
+Hardware can require an object
 referenced by an \tcode{atomic_ref}
 to have stricter alignment\iref{basic.align}
 than other objects of type \tcode{T}.
 Further, whether operations on an \tcode{atomic_ref}
-are lock-free could depend on the alignment of the referenced object.
-For example, lock-free operations on \tcode{std::complex<double>}
-could be supported only if aligned to \tcode{2*alignof(double)}.
+are lock-free can depend on the alignment of the referenced object.
+For example, it is possible for
+operations on a \tcode{std::complex<double>} object
+to be lock-free only if the object is aligned to \tcode{2*alignof(double)}.
```

Comment:
This seems to be ambiguous now. "It is possible for X only if Y" can be read as "if and only if Y".

This ISO rule is stupid.

How about "For example, operations on `std::complex<double>` objects can be lock-free if aligned to `2*alignof(double)` but the hardware can require a lock when aligned to `alignof(double)`."

I really thing "might" is a better word for what we're trying to say.

Reply:
Completely agree about the stupidity of the rule. But I have argued, my face is blue, and we got no closer to publishing the standard. :(

I don't like using "can" for possibility rather than ability or capability. How about this:

"For example, there might exist hardware that supports lock-free atomic operations on `std::complex<double>` objects only if those objects are aligned to `2*alignof(double)`."

Reply:
That uses "might", which are also being replaced.

Reply:
> I don't like using "can" for possibility rather than ability or capability.

@zygoloid : Sorry, now I'm confused. Your initial mail gave the guidance "Use "can" when referring to ability or possibility in all cases." This clearly says "possibility", it seems.

Reply:
You're right, that's what ISO/CS wanted. Doesn't mean that I necessarily like the outcome though :)

@johelegp Curses. But I think that's fairly easy to fix: "For example, it is possible that the hardware supports lock-free atomic operations on `std::complex<double>` objects only if those objects are aligned to `2*alignof(double)`."

Reply:
fixed

Reply:
It now mixes plural and singular. It starts talking about objects (plural) then says "only if the object" (singular). Richard's "only if those objects are aligned" seems right to me.

Apart from that, this looks good now.

Reply:
Good point, but we should go singular instead. Fixed.